### ISSUE TITLE
[Feature Request] Settings should be overridable from .atom-build.json

### ISSUE THREAD

**Issue by forivall** [noseglid/atom-build#103]
Ideally, the toplevel of `.atom-build.json` shouldn't be a build path, but instead be settings to use and override the default settings. Or, these settings could be stored in some other file like `.atom-build.user.json`.

My use case is that, I want my c++ projects to build on save, but not everything else.

**Comment by noseglid** [comment]
Not sure what you're asking for here. Do you want to override the global settings on a per-project basis? Why do you prefer your `C++` projects to build on save, and not other projects?

**Comment by forivall** [comment]
Exactly, the ability to override global settings on a per-project basis.

To be specific, I don't really want atom and node projects to `npm install` every time I save a file, it seems like more work than necessary. Unless npm has updated recently so that it checks which packages are already installed first, but last time i checked, it does a full install every time. And npm install isn't really checking my code, whereas my c++ build is checking my code for errors every time. If the build was running unit tests, then i'd maybe want it to build on every save, but I'd use a custom build file for that. And then with that custom build file, I could enable build on save.

It's just an idea. If it doesn't appeal to you, I'd be fine with writing and submitting a PR for this.

**Comment by noseglid** [comment]
Perhaps the discussion should rather be if these configurations should be on a per-project basis, rather than global. But then you wouldn't be able to use the built in tools (since it would require creating an `.atom-build.json`-file to be able to set the configurations) if you wanted `Build on Save` or `Save on Build`.

**Comment by averrin** [comment]
:+1: 
I use "gulp watch" for JS (no autobuild needed), but i want "Build on Save" for my golang project.

**Comment by paulequilibrio** [comment]
I need this feature, too.
It is possible that the global settings can act as the default settings, if none is found in the `.atom-build.json`?

**Comment by paulequilibrio** [comment]
I need this feature, too.
It is possible that the global settings can act as the default settings, if none is found in the `.atom-build.json`?
